pkg/hub/controllers/cluster/reconciler.go

```diff
@@ -286,10 +286,25 @@ func (r *Reconciler) updateNodeIps(ctx context.Context, cr *hubv1alpha1.Cluster)
 	})
 	if err != nil {
 		log.Error(err, "Error updating to node ip's on hub cluster")
+		err := r.EventRecorder.RecordEvent(ctx, &events.Event{
```

Comment:
why is this event needed? 
Failing to update is a normal error in controller's pattern,
This can give a wrong interpretation to user , because in next reconcile loop it can succeed.
And this is an "internal" event - details should not be exposed to user!

Reply:
We have discussed this case, so we want to raise an event if there is a failure and in the next reconcile loop if the update is successful we raise a new event that now update has happened. So the latest event will be successful.